File: 2 CPP/Leetcode-1266-MinimumTimeToVisitAllPoints.cpp
class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int ans = 0;
        for(int i=0;i<points.size()-1;++i){
            vector <int> pt = points[i];
            vector <int> npt = points[i+1];
            int xdiff = abs(npt[0]-pt[0]);
            int ydiff = abs(npt[1]-pt[1]);
            ans+=max(xdiff,ydiff);
        }
        return ans;
    }
};

File: 1 Python/Leetcode-1266-MinimumTimeToVisitAllPoints.py
class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        time = 0
        for i in range(len(points)-1):
            pt = points[i]
            npt = points[i+1]
            xdiff = abs(npt[0]-pt[0])
            ydiff = abs(npt[1]-pt[1])
            time+=max(xdiff,ydiff)
        return time
